tracer/src/Datadog.Tracer.Native/debugger_rejit_preprocessor.cpp

```diff
@@ -250,9 +276,116 @@ void DebuggerRejitPreprocessor::UpdateMethod(RejitHandlerModuleMethod* methodHan
 {
     const auto debuggerMethodHandler = dynamic_cast<DebuggerRejitHandlerModuleMethod*>(methodHandler);
     debuggerMethodHandler->AddProbe(probe);
-    ProbesMetadataTracker::Instance()->AddMethodToProbe(probe->probeId,
-        debuggerMethodHandler->GetModule()->GetModuleId(),
-        debuggerMethodHandler->GetMethodDef());
+    ProbesMetadataTracker::Instance()->AddMethodToProbe(
+        probe->probeId, debuggerMethodHandler->GetModule()->GetModuleId(), debuggerMethodHandler->GetMethodDef());
+}
+
+void DebuggerRejitPreprocessor::EnqueueNewMethod(const MethodProbeDefinition& definition,
+                                                 ComPtr<IMetaDataImport2>& metadataImport,
+                                                 ComPtr<IMetaDataEmit2>& metadataEmit, const ModuleInfo& moduleInfo,
+                                                 mdTypeDef typeDef, std::vector<MethodIdentifier>& rejitRequests,
+                                                 unsigned methodDef, FunctionInfo functionInfo,
+                                                 RejitHandlerModule* moduleHandler)
+{
+    const auto [newMethodDef, newFunctionInfo] = DebuggerRejitPreprocessor::TransformKickOffToMoveNext(
+        metadataImport, metadataEmit, moduleHandler, typeDef, methodDef, functionInfo);
+    RejitPreprocessor::EnqueueNewMethod(definition, metadataImport, metadataEmit, moduleInfo, typeDef, rejitRequests,
+                                        newMethodDef, newFunctionInfo, moduleHandler);
+}
+
+std::tuple<mdMethodDef, FunctionInfo> DebuggerRejitPreprocessor::TransformKickOffToMoveNext(
```

Comment:
Not a must have, but this method is pretty long and actually and does two distinct separate things. It might be worthwhile to extract out two methods: `GetMoveNextMethodFromKickOffMethod` and `TransformMoveNextMethod`.

Reply:
Fixed in 871a652d9b150d63443d87a542e34d4643145adf